src/expression/definitions/index_of.ts

```diff
@@ -60,16 +60,28 @@ class IndexOf implements Expression {
             throw new RuntimeError(`Expected first argument to be of type boolean, string, number or null, but found ${toString(typeOf(needle))} instead.`);
         }
 
-        if (!isValidNativeType(haystack, ['string', 'array'])) {
-            throw new RuntimeError(`Expected second argument to be of type array or string, but found ${toString(typeOf(haystack))} instead.`);
+        let fromIndex;
+        if (this.fromIndex) {
+            fromIndex = (this.fromIndex.evaluate(ctx) as number);
         }
 
-        if (this.fromIndex) {
-            const fromIndex = (this.fromIndex.evaluate(ctx) as number);
+        if (isValidNativeType(haystack, ['string'])) {
+            if (needle === '') {
```

Comment:
Can you clarify this? won't we get the same results if we remove this if (and the return 0)?

Reply:
Oh you’re right, I changed it to stop using `Array.prototype.indexOf` at the last minute, so this is no longer necessary. 👍